NuGet/ChustaSoft.Tools.Authorization/Services/UserService.cs

```diff
@@ -215,6 +225,37 @@ public void Review(TUser user)
                 user.Culture = _authorizationSettings.DefaultCulture;
         }
 
+        public async Task<SignInResult> ExternalSignInAsync(bool isPersistent)
+        {
+            var loginInfo = await _signInManager.GetExternalLoginInfoAsync();
+            return await _signInManager.ExternalLoginSignInAsync(loginInfo.LoginProvider, loginInfo.ProviderKey, isPersistent);
+        }
+
+        public async Task CreateExternalAsync(string defaultRole)
```

Comment:
Return TUser as the normal Create for cohession with the other similar functionality

Reply:
It's true that for better cohession we should be returning here a TUser object as we have in the other SignIn methods. In that case, we return SigninResult because we are first trying to signin the user, if the user does not exists on db it will have a Failed result, then we go to create the user and then we perform the signin. This way we can have one less call to usermanager, because we are not checking first if the user exists on db.

Therefore, I see some options here:

- Change the signature of the method and return a TUser, if there is a null it means we need to create the user (with this we will be losing all the other possible values for the SigninResult, the user could be locked out for instance, or not allowed, but maybe we don't want to take care of this information, just ok or not ok, as we do for the SignByEmail or SignByPhone method)
- Create an ExternalUSerService and move this logic there
- Any other suggestion is welcomed

Reply:
Hi @vvillena !!

My comment was related to CreateExternalAsync method, not ExternalSignInAsync,

In CreateExternalAsync we are just defining the Task with void return, if we align with the basic user creation, after that we are sending back the TUser, but in that case we are avoing this info to the thread of invokation, and in fact, inside the block of CreateExternalAsync , in line 243, the TUser is created and can potentially be sent back

Reply:
Understood! now the user is returned.